CUDADataFormats/Track/interface/PixelTrackUtilities.h

```diff
@@ -0,0 +1,241 @@
+#ifndef CUDADataFormats_Track_PixelTrackUtilities_h
+#define CUDADataFormats_Track_PixelTrackUtilities_h
+
+#include <Eigen/Dense>
+#include <Eigen/Core>
+#include "Geometry/CommonTopologies/interface/SimplePixelTopology.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/HistoContainer.h"
+#include "DataFormats/SoATemplate/interface/SoALayout.h"
+
+namespace pixelTrackSoA {
+
+  enum class Quality : uint8_t { bad = 0, edup, dup, loose, strict, tight, highPurity, notQuality };
+  constexpr uint32_t qualitySize{uint8_t(Quality::notQuality)};
+  const std::string qualityName[qualitySize]{"bad", "edup", "dup", "loose", "strict", "tight", "highPurity"};
+  inline Quality qualityByName(std::string const &name) {
+    auto qp = std::find(qualityName, qualityName + qualitySize, name) - qualityName;
+    return static_cast<Quality>(qp);
+  }
+
+}  // namespace pixelTrackSoA
+
+template <typename TrackerTraits>
+struct TrackSoA {
+  static constexpr int32_t S = TrackerTraits::maxNumberOfTuples;
+  static constexpr int32_t H = TrackerTraits::avgHitsPerTrack;
+  // Aliases in order to not confuse the GENERATE_SOA_LAYOUT
+  // macro with weird colons and angled brackets.
+  using Vector5f = Eigen::Matrix<float, 5, 1>;
+  using Vector15f = Eigen::Matrix<float, 15, 1>;
+  using Quality = pixelTrackSoA::Quality;
+
+  using hindex_type = uint32_t;
+
+  using HitContainer = cms::cuda::OneToManyAssoc<hindex_type, S + 1, H * S>;
```

Comment:
I would avoid using the `S` and `H` aliases and use directly `TrackerTraits::maxNumberOfTuples` and `TrackerTraits::avgHitsPerTrack`.

I think this is a matter of taste, and a very minor thing anyway, so do use what you prefer.